src/Filesystem.php

```
@@ -350,7 +372,22 @@ public function getMetadata($path)
         $path = Util::normalizePath($path);
         $this->assertPresent($path);
 
-        return $this->getAdapter()->getMetadata($path);
+        // Deal with inconsistent adapter behavior.
+        // Some adapters will throw exceptions for directories; Some return false; Some return an array
+        try {
+            $metaData = $this->getAdapter()->getMetadata($path);
+        } catch (\Exception $e) {
+            // Do nothing
+        }
+
+        // If file exists, but returns no metadata, assume it's a directory
+        if (empty($metaData)) {
+            $metaData = [
+                'type' => 'dir',
+                'path' => $path,
+            ];
+        }
+        return $metaData;
```

This will swallow all sorts of exceptions that could be thrown by an adapter that the developer should be aware of.